### PR TITLE
Update OpenAPI

### LINKED ISSUE #35: OpenAPI
The OpenAPI document needs to be updated.

This is something I will fix.

### PR DESCRIPTION
**Pull request by Cactu5**
This pull request fixes #35

### CODE REVIEW

File: OpenAPI/decentralized-iso20022-v1.yaml [hunk 1]
```diff
@@ -1,152 +1,395 @@
 openapi: 3.0.3
 info:
   title: Decentralized ISO 20022 Communication Service
-  description: This is the API specificaiton for the APIs in the decentralized ISO 20022 communication service.
+  description: This is the API specificaiton for the external API in the decentralized ISO 20022 communication service.
```
Comment (william-stacken) [inline OpenAPI/decentralized-iso20022-v1.yaml:4]: Why was this changed to external API?

File: OpenAPI/decentralized-iso20022-v1.yaml [hunk 2]
```diff
@@ -1,152 +1,395 @@
...
-                  type: string
-
-  /api/v1/pacs:
-    post:
-      tags:
-        - External
+        - Internal/External
       summary: Receive a pacs message from another financial institution.
-      description: The endpoint from which the microservice receives pacs messages from another financial institution.<br><br>A valid pacs message should be received with a header head.001.001.02.
+      description: <strong>Internal:</strong> The endpoint where the financial institution sends the pacs.008.001.09 message to the microservice for validation before forwarding it to the recieving financial institution specified in the 'to' element in the 'AppHdr' element. <br><br><strong>External:</strong> The endpoint from which the microservice receives pacs messages from another financial institution. A valid pacs message should be received with a header head.001.001.02.
```
Comment (william-stacken) [inline OpenAPI/decentralized-iso20022-v1.yaml:19]: I noticed that the term was used before too, but I think it should be changed to "communication service" since that is what it is referred to as in a bunch of other places.

File: OpenAPI/decentralized-iso20022-v1.yaml [hunk 3]
```diff
@@ -1,152 +1,395 @@
...
+          example: 2021/05/12 13:56:41
+        status:
+          type: integer
+          example: 403
+        code: 
+          type: string
+          example: FORBIDDEN
+        message:
+          type: string
+          example: a message describing the problem
```
Comment (william-stacken) [inline OpenAPI/decentralized-iso20022-v1.yaml:292]: I got the impression of the example message for 400 Bad Request that it should be an example that actually shows up in use, but here in 403 Forbidden it sounds like a description of the message that will show up in use. Maybe it should be synchronized so that all examples are one or the other?

**Comment by william-stacken** [comment]
Also, if you could revert the repository to checkout to in the workflow it should fix the pipeline. I realized I cannot since I do not have push access to this repo.

File: OpenAPI/decentralized-iso20022-v1.yaml (hunk 1 above)
Comment (Cactu5) [reply OpenAPI/decentralized-iso20022-v1.yaml:4]: This was initially changed when I was going to do a separate document for external and internal but realized that the API has the same structure. Will change this back. 👍

File: OpenAPI/decentralized-iso20022-v1.yaml (hunk 2 above)
Comment (Cactu5) [reply OpenAPI/decentralized-iso20022-v1.yaml:19]: Sounds good, will change that!

File: OpenAPI/decentralized-iso20022-v1.yaml (hunk 3 above)
Comment (Cactu5) [reply OpenAPI/decentralized-iso20022-v1.yaml:292]: The problem is that I couldn't figure how what message goes for what error. Could you send me the messages perhaps?

**Comment by Cactu5** [comment]
> Also, if you could revert the repository to checkout to in the workflow it should fix the pipeline. I realized I cannot since I do not have push access to this repo.

I pushed a fix yesterday for the workflow and merged the latest main into this branch. The workflow works now. 🙂

File: OpenAPI/decentralized-iso20022-v1.yaml (hunk 3 above)
Comment (william-stacken) [reply OpenAPI/decentralized-iso20022-v1.yaml:292]: An example for 403 is `message.bad-internal-cert`. You can check the controller classes as that's where most of the `ResponseStatusException`s are thrown. The validation service classes also keep some of the lower level error messages for when messages are invalid.

Comment (Cactu5) [reply OpenAPI/decentralized-iso20022-v1.yaml:292]: Thanks! Found them!

**Comment by william-stacken** [comment]
LGTM! Feel free to merge! 🚀
